metricshub-hardware/src/main/java/org/metricshub/hardware/util/MonitorNameBuilder.java

```diff
@@ -0,0 +1,1020 @@
+	 * </p>
+	 *
+	 * @param type                  the type prefix to be added (maybe {@code null})
+	 * @param displayId             the primary display name for the device
+	 * @param deviceId              the device identifier to use as a fallback name
+	 * @param idCount               the ID count (must be non-null)
+	 * @param trimPattern           a {@link Pattern} (regex) used to remove unwanted words from the device ID;
+	 *                              may be {@code null}
+	 * @param additionalLabelFields optional additional label fields to append inside parentheses
+	 * @return the full name following the standard naming convention
+	 */
+	public static String buildName(
+		final String type,
+		final String displayId,
+		final String deviceId,
+		@NonNull final String idCount,
+		final Pattern trimPattern,
+		final String... additionalLabelFields
+	) {
+		final StringBuilder fullName = new StringBuilder();
+
+		// Add the type if provided
+		if (hasMeaningfulContent(type)) {
+			fullName.append(type).append(": ");
+		}
+
+		// Determine the base name to use
+		String name = null;
+		if (hasMeaningfulContent(displayId)) {
+			name = displayId;
+		} else if (hasMeaningfulContent(deviceId)) {
+			name = (trimPattern == null) ? deviceId : trimPattern.matcher(deviceId).replaceAll("");
+			if (name.length() > ID_MAX_LENGTH) {
+				name = idCount;
+			}
+		}
+		// Fallback to idCount if no meaningful name is found
+		if (!hasMeaningfulContent(name)) {
+			name = idCount;
+		}
+		fullName.append(name);
+
+		// Append additional label information if available
+		final String additionalLabel = joinWords(additionalLabelFields);
+		if (hasMeaningfulContent(additionalLabel)) {
+			fullName.append(" (").append(additionalLabel).append(")");
+		}
+		return trimUnwantedCharacters(fullName.toString());
+	}
+
+	/**
+	 * Builds the battery name based on the current implementation.
+	 *
+	 * @param monitor the {@link Monitor} instance representing the battery device
+	 * @return the formatted battery name
+	 */
+	public static String buildBatteryName(final Monitor monitor) {
+		// Check the attributes
+		final Map<String, String> attributes = monitor.getAttributes();
+		Assert.notNull(attributes, ATTRIBUTES_CANNOT_BE_NULL);
+
+		// Build the name
+		return buildName(
+			null,
+			attributes.get(DISPLAY_ID),
+			attributes.get(DEVICE_ID),
+			attributes.get(ID_COUNT),
+			BATTERY_TRIM_PATTERN,
+			joinVendorAndModel(attributes),
+			attributes.get(BATTERY_TYPE)
+		);
+	}
+
+	/**
+	 * Builds the blade name based on the current implementation.
+	 *
+	 * @param monitor the {@link Monitor} instance representing the blade device
+	 * @return the formatted blade name
+	 */
+	public static String buildBladeName(final Monitor monitor) {
+		// Check the attributes
+		final Map<String, String> attributes = monitor.getAttributes();
+		Assert.notNull(attributes, ATTRIBUTES_CANNOT_BE_NULL);

```

Comment:
Remove all the Assert.nonNull on the attributes. By design, `attributes` can never be null on a Monitor.